modules/sdk-core/src/bitgo/tss/eddsa/eddsa.ts

```diff
@@ -373,3 +376,33 @@ export async function getTSSSignature(
   const rawSignature = Buffer.concat([Buffer.from(signature.R, 'hex'), Buffer.from(signature.sigma, 'hex')]);
   return rawSignature;
 }
+
+/**
+ * Verifies that a TSS wallet signature was produced with the expected key and that the signed data contains the
+ * expected common keychain, the expected user and backup key ids as well as the public share that is generated from the
+ * private share that was passed in.
+ */
+export async function verifyWalletSignature(params: {
+  walletSignature: openpgp.Key;
+  bitgoPub: openpgp.Key;
+  commonKeychain: string;
+  userKeyId: string;
+  backupKeyId: string;
+  decryptedShare: string;
+  verifierIndex: 1 | 2; // the index of the verifier, 1 means user, 2 means backup
+}): Promise<void> {
+  const rawNotations = await commonVerifyWalletSignature(params);
+
+  const { decryptedShare, verifierIndex } = params;
+
+  const publicShare =
+    Buffer.from(await sodium.crypto_sign_ed25519_sk_to_pk(Buffer.from(decryptedShare.slice(0, 64), 'hex'))).toString(
```

Comment:
please use the function crypto_scalarmult_ed25519_base_noclamp instead of crypto_sign_ed25519_sk_to_pk.